caom2-artifact-resolvers/src/main/java/ca/nrc/cadc/caom2/artifact/resolvers/GeminiResolver.java

```diff
@@ -77,16 +77,22 @@ private String getPath(URI uri) {
             throw new IllegalArgumentException("Malformed URI. Expected 2 path components, found " + path.length);
         }
 
-        String requestType = path[0];
-        if (!(requestType.equals(FILE_URI) || requestType.equals(PREVIEW_URI))) {
-            throw new IllegalArgumentException("Invalid URI. Expected 'file' or 'preview' and got " + requestType);
+        String archive = path[0];
+        if (!(archive.equals(ARCHIVE))) {
+            throw new IllegalArgumentException("Invalid URI. Expected archive: " + ARCHIVE + ", actual archive: " + archive);
         }
 
         String fileName = path[1];
+        String fileType = FILE_URI;
+        String[] fileNameParts = fileName.split("\\.");
+        if (fileNameParts[1].equals("jpg")) {
```

Comment:
Here I might just go with fileName.endsWith(".jpg") so that other dots in the fileName won't break this code...